Software/bsw/static/Mcal/CAN/src/Can.c

```diff
@@ -1343,163 +1344,71 @@ Std_ReturnType Can_SetControllerMode( uint8 Controller, Can_ControllerStateType
                 if(DisableCnt[0U]==0U)
```

Comment:
For the the disable counter I believe the following:
Stopped request:
Stopping steps should be execute once and only the counter will be only incremented afterwards.
Starting request:
The counter should be decremented first and we check if we reach zero, only then we enable the CAN.